### DISCUSSION TITLE
Update XUnit V3

### DISCUSSION THREAD

**Discussion by stefannikolei** [SixLabors/ImageSharp#2853] [Ideas]
Theres an update for xunit. 

I think following steps are necessary:

1: Update SharedInfrastructure

2: Update All Repos

- Imagesharp
- Imagesharp.Web
- Imagesharp.Drawing
- Fonts

Did i miss any?

If you want i can tackle this.

**Comment by antonfirsov** [comment]
I think the step 0 should be an experiment changing SharedInfrastructure+ImageSharp together to see if it everything works well or if there are breaking changes causing headache. This would need branching SharedInfrastructure and referencing that branch from the submodule in an ImageSharp experimental branch. Then, I would open a draft PR in ImageSharp to see if test runs are passing in the CI. Afterwards, we can repeat the same experiment with the remaining 3 libraries, go ahead with the SharedInfrastructure PR, then update and merge the library PR-s.

**Reply by stefannikolei** [reply]
> I think the step 0 should be an experiment changing SharedInfrastructure+ImageSharp together to see if it everything works well or if there are breaking changes causing headache. This would need branching SharedInfrastructure and referencing that branch from the submodule in an ImageSharp experimental branch. Then, I would open a draft PR in ImageSharp to see if test runs are passing in the CI. Afterwards, we can repeat the same experiment with the remaining 3 libraries, go ahead with the SharedInfrastructure PR, then update and merge the library PR-s.

Ok. Would you create a branch for me that I can reference in SharedInfrastruxture?

**Reply by JimBobSquarePants** [reply]
Do you still need that branch?

**Reply by stefannikolei** [reply]
No I am good. found a way to test it. 

Got really far but struggled on one place. Gonna try to write it up. Was something with an own implementation of an xunit interface to get all test files. 

I try to write it up. To discuss it and perhaps get some help to port/fix it 😅
